core/src/main/java/hudson/util/spring/BeanBuilder.java

```diff
@@ -1,652 +0,0 @@
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package hudson.util.spring;
-
-import groovy.lang.Binding;
-import groovy.lang.Closure;
-import groovy.lang.GString;
-import groovy.lang.GroovyObject;
-import groovy.lang.GroovyObjectSupport;
-import groovy.lang.GroovyShell;
-import groovy.lang.MetaClass;
-import groovy.lang.MissingMethodException;
-import org.apache.commons.lang.ArrayUtils;
-import org.codehaus.groovy.control.CompilerConfiguration;
-import org.codehaus.groovy.runtime.DefaultGroovyMethods;
-import org.codehaus.groovy.runtime.InvokerHelper;
-import org.springframework.beans.factory.config.BeanDefinition;
-import org.springframework.beans.factory.config.RuntimeBeanReference;
-import org.springframework.beans.factory.support.ManagedList;
-import org.springframework.beans.factory.support.ManagedMap;
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.support.StaticApplicationContext;
-import org.springframework.core.io.Resource;
-import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
-import org.springframework.web.context.WebApplicationContext;
-
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.ListIterator;
-import java.util.Map;
-import java.util.Map.Entry;
-
-/**
- * <p>Runtime bean configuration wrapper. Like a Groovy builder, but more of a DSL for
- * Spring configuration. Allows syntax like:</p>
- *
- * <pre>
- * import org.hibernate.SessionFactory
- * import org.apache.commons.dbcp.BasicDataSource
- *
- * BeanBuilder builder = new BeanBuilder()
- * builder.beans {
- *   dataSource(BasicDataSource) {                  // ← invokeMethod
- *      driverClassName = "org.hsqldb.jdbcDriver"
- *      url = "jdbc:hsqldb:mem:grailsDB"
- *      username = "sa"                            // ← setProperty
- *      password = ""
- *      settings = [mynew:"setting"]
- *  }
- *  sessionFactory(SessionFactory) {
- *  	   dataSource = dataSource                 // ← getProperty for retrieving refs
- *  }
- *  myService(MyService) {
- *      nestedBean = { AnotherBean bean-&gt;          // ← setProperty with closure for nested bean
- *      		dataSource = dataSource
- *      }
- *  }
- * }
- * </pre>
- * <p>
- *   You can also use the Spring IO API to load resources containing beans defined as a Groovy
- *   script using either the constructors or the loadBeans(Resource[] resources) method
- * </p>
- *
- * @author Graeme Rocher
- * @since 0.4
- *
- */
-public class BeanBuilder extends GroovyObjectSupport {
```

Comment:
Deleting this whole package; now constructing security objects directly in Java code. This means dropping support for the dubious feature of overriding Jenkins’ security architecture by replacing a file in the WAR; you would now have to use supported configuration settings only.